cli/internal/cmd/upgradecheck.go

```diff
@@ -235,9 +235,10 @@ func (u *upgradeCheckCmd) upgradeCheck(cmd *cobra.Command, fetcher attestationco
 		return fmt.Errorf("planning terraform migrations: %w", err)
 	}
 	defer func() {
-		// Remove the upgrade directory
-		if err := u.fileHandler.RemoveAll(upgradeDir); err != nil {
-			u.log.Debugf("Failed to clean up Terraform migrations: %s", err)
+		// User doesn't expect to see any changes in his workspace after an "upgrade plan",
+		// therefore, roll back to the backed up state.
+		if err := u.terraformChecker.RollbackClusterWorkspace(); err != nil {
+			u.log.Debugf("Failed to rollback Terraform workspace: %s", err)
```

Comment:
Same here.
Since we now write directly into the terraform workspace, we need to tell the user directly if restoring the backup failed.
Just printing it in the debug log is not enough.